src/main/java/org/openstreetmap/atlas/checks/validation/tag/StreetNameIntegersOnlyCheck.java

```diff
@@ -0,0 +1,105 @@
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.names.NameLeftTag;
+import org.openstreetmap.atlas.tags.names.NameRightTag;
+import org.openstreetmap.atlas.tags.names.NameTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check flags {@link Edge}s that are car navigable highways and have a name tag that contains
+ * only integers.
+ *
+ * @author bbreithaupt
+ */
+public class StreetNameIntegersOnlyCheck extends BaseCheck
+{
+
+    private static final long serialVersionUID = 3439708862406928654L;
+
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList("Street {0,number,#} has a name containing only integers.");
+
+    private static final List<String> NAME_KEYS_DEFAULT = Arrays.asList(NameTag.KEY,
+            NameLeftTag.KEY, NameRightTag.KEY);
+
+    private final List<String> nameKeys;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public StreetNameIntegersOnlyCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.nameKeys = (List<String>) configurationValue(configuration, "name.keys.filter",
+                NAME_KEYS_DEFAULT);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return HighwayTag.isCarNavigableHighway(object)
+                && !this.isFlagged(object.getOsmIdentifier()) && object instanceof Edge
+                && this.nameKeys.stream()
+                        .anyMatch(nameKey -> object.getOsmTags().containsKey(nameKey));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        // Try to convert name to an integer. If it converts without failure it should be flagged.
+        for (final String nameKey : this.nameKeys)
+        {
+            final Optional<String> nameValue = object.getTag(nameKey);
+            if (nameValue.isPresent())
+            {
+                try
+                {
+                    Integer.parseInt(nameValue.get().replaceAll(" ", ""));
```

Comment:
Can you test and make sure this will replace tab and other possible unicode whitespace characters? https://www.fileformat.info/info/unicode/category/Zs/list.htm 

If not can you update it to?